README.rst

```diff
@@ -453,207 +517,252 @@ Navigating the Containers
-  ``i`` is a token.  This function returns the token of the
+  ``(m,s)`` is a token.  This function returns the semitoken of the
   previous entry in the container according to the sort order of the
-  keys.  If ``i`` indexes the first item, this routine returns the before-start
-  token.  It is an error to invoke this function if ``i`` is the
-  before-start token.  If ``i`` is the past-end token, then this
-  routine returns the token of the last item in the sort order (i.e., the
-  same token returned by the ``endof`` function).
+  keys.  If ``(m,s)`` indexes the first item, this routine returns the before-start
+  semitoken.  It is an error to invoke this function if ``(m,s)`` is the
+  before-start token.  If ``(m,s)`` is the past-end token, then this
+  routine returns the smitoken of the last item in the sort order (i.e., the
+  same semitoken returned by the ``endof`` function).  
   Time: O(log *n*)
 
 ``searchsortedfirst(m,k)``
-  Argument ``m`` is a SortedDict and
-  ``k`` is an element of the key type.  This routine returns the token
+  Argument ``m`` is a SortedDict, SortedMultiDict or SortedSet and
+  ``k`` is an element of the key type.  This routine returns the semitoken
   of the first item in the container whose key is greater than or equal to
-  ``k``.  If there is no such key, then the past-end token
+  ``k``.  If there is no such key, then the past-end semitoken
   is returned.
   Time: O(*c* log *n*)
 
 ``searchsortedlast(m,k)``
-  Argument ``m`` is a SortedDict and
-  ``k`` is an element of the key type.  This routine returns the token
+  Argument ``m`` is a SortedDict, SortedMultiDict or SortedSet and
+  ``k`` is an element of the key type.  This routine returns the semitoken
   of the last item in the container whose key is less than or equal to
-  ``k``.  If there is no such key, then the before-start token
+  ``k``.  If there is no such key, then the before-start semitoken
   is returned.
   Time: O(*c* log *n*)
 
+
 ``searchsortedafter(m,k)``
-  Argument ``m`` is a SortedDict and
-  ``k`` is an element of the key type.  This routine returns the token
+  Argument ``m`` is a SortedDict, SortedMultiDict or SortedSet and
+  ``k`` is an element of the key type.  This routine returns the semitoken
   of the first item in the container whose key is greater than
-  ``k``.  If there is no such key, then the past-end token
+  ``k``.  If there is no such key, then the past-end semitoken
   is returned.
   Time: O(*c* log *n*)
 
 
+``searchequalrange(m,k)``
+   Argument ``m`` is a SortedMultiDict and ``k`` is an element of the
+   key type.  This routine returns a pair of semitokens; the first 
+   of the pair is the semitoken addressing the first item in the container
+   with key ``k`` and the second is the semitoken addressing one past the
+   last item in the container with key ``k``.  If the two semitokens returned
+   are equal, this means that no items in the container matched key ``k``.
+   Time: O(*c* log *n*)
```

Comment:
Container (well, array) ranges in Julia are typically `first_idx:last_idx`.  For consistency, it might be better to have the second semitoken refer to the last element, rather than the element *after* the last element.  

I'm not sure how this would affect other parts of the code, however.